### PR TITLE
Add a liveness and readiness probe to deck container

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/assign @cjwagner @fejta @Katharine 
/cc @BenTheElder @krzyzacy

### CODE REVIEW

File: prow/cluster/deck_deployment.yaml [hunk 1]
```diff
@@ -80,4 +92,4 @@ spec:
           name: config
       - name: job-config
         configMap:
-          name: job-config
+          name: job-config
```
Comment (krzyzacy) [inline prow/cluster/deck_deployment.yaml]: babynit: new line

File: prow/cluster/deck_deployment.yaml [hunk 2]
```diff
@@ -64,6 +64,18 @@ spec:
         - name: job-config
           mountPath: /etc/job-config
           readOnly: true
+        livenessProbe:
```
Comment (fejta) [inline prow/cluster/deck_deployment.yaml]: Isn't this code new? If so we need to wait until its on a version that has it

Comment (stevekuznetsov) [reply prow/cluster/deck_deployment.yaml]: As per the tracking issue I believe it's already deployed -- could you please confirm?

File: prow/cluster/deck_deployment.yaml (hunk 1 above)
Comment (stevekuznetsov) [reply prow/cluster/deck_deployment.yaml]: :man_facepalming: editors..

**Comment by stevekuznetsov** [comment]
@krzyzacy that's the polling period, not the timeout. Added a 30s timeout for the readiness -- I can't tell from my logs how long it takes to start up?

File: prow/cluster/deck_deployment.yaml (hunk 2 above)
Comment (fejta) [reply prow/cluster/deck_deployment.yaml]: This file says what commit is deployed: gcr.io/k8s-prow/deck:v20190307-fd53164

Is your change in that?

Comment (stevekuznetsov) [reply prow/cluster/deck_deployment.yaml]: Oops, messed up my check -- it's barely not. Do you want to bump to grab it? You could also do both in one PR and not incur one slow 502-ridden rollout

Comment (fejta) [reply prow/cluster/deck_deployment.yaml]: I'll want to bump again today once we get my tide fix and the config PR in, yes

**Comment by krzyzacy** [comment]
> I can't tell from my logs how long it takes to start up?

@Katharine do we have some metrics? is this fine?

**Comment by Katharine** [comment]
> > I can't tell from my logs how long it takes to start up?
> 
> @Katharine do we have some metrics? is this fine?

We don't have metrics, but when everything is restarting we're basically blocked on how long tide takes to become responsive. In my anecdotal experience, you'd probably want a 5-10 minute timeout to be safe. Thirty seconds is almost certainly not long enough.

(On the flip side, I think even if it keeps being restarted, because we're blocked on _tide_ rather than anything deck does, restarting deck repeatedly isn't the worst thing. But probably also not the ebst one.)

**Comment by fejta** [comment]
/hold
blocked on https://github.com/kubernetes/test-infra/issues/11690

**Comment by stevekuznetsov** [comment]
That got resolved (lol)
/hold cancel

**Comment by stevekuznetsov** [comment]
Changed to a 10min readiness timeout

**Comment by stevekuznetsov** [comment]
@fejta this now has pre-requisites deployed

**Comment by fejta** [comment]
Will this hopefully make it where deck restarts don't cause outages?

**Comment by fejta** [comment]
Deploying https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/post-test-infra-deploy-prow/1218

**Comment by stevekuznetsov** [comment]
Yes, this should ensure working replicas of `deck` do not get stopped before a new replica is both healthy _and_ ready

**Pull request merged**